addOns/client/src/main/java/org/zaproxy/addon/client/ExtensionClientIntegration.java

```diff
@@ -884,10 +884,12 @@ public void sessionModeChanged(Mode mode) {
         }
     }
 
-    public void exportClientMap(String path) {
+    public boolean exportClientMap(String path) {
```

Comment:
Add an input boolean like `isApi` then at `if (hasView())` update to `if (hasView() && !isApi)` then the dialog won't be triggered by an API interaction.

Could also add `public boolean exportClientMap(String path)` to just `return exportClientMap(path, false);` so that the existing call still behaves the existing way.